src/mcx_utils.c

```diff
@@ -196,7 +196,7 @@ const char boundarydetflag[] = {'0', '1', '\0'};
 
 const char* srctypeid[] = {"pencil", "isotropic", "cone", "gaussian", "planar",
                            "pattern", "fourier", "arcsine", "disk", "fourierx", "fourierx2d", "zgaussian",
-                           "line", "slit", "pencilarray", "pattern3d", "hyperboloid", ""
+                           "line", "slit", "pencilarray", "pattern3d", "hyperboloid", "ring", ""
```

Comment:
there are a few other places where these source type constants are defined, you will have to run `cd mcx/src; grep 'zgaussian' -R *` and modify all other units. I believe `pmcx.cpp` and `mcxlab.cpp` both need to be updated.